src/Parsers/ASTSystemQuery.cpp

```diff
@@ -39,6 +40,58 @@ const char * ASTSystemQuery::typeToString(Type type)
     return type_name.data();
 }
 
+String ASTSystemQuery::getDatabase() const
+{
+    String name;
+    tryGetIdentifierNameInto(database, name);
+    return name;
+}
+
+String ASTSystemQuery::getTable() const
+{
+    String name;
+    tryGetIdentifierNameInto(table, name);
+    return name;
+}
+
+void ASTSystemQuery::setDatabase(const String & name)
+{
+    if (name.empty() && !database)
+        return;
+
+    assert(!name.empty());
```

Comment:
For what case first `if` is for? Not obvious why this invariant `!name.empty()` should always be true here because of the case above.

Reply:
The case is that we cannot create an empty `ATRIdentifier` or assign an empty name to it. If we want to assign an empty name after we assigned non-empty value (which seems to be a case basically absent in current code base) then we will need to find that ASTPtr in our children vector and then delete it which is `O(n)` where `n` is `children.size()`. Seems not really good, but can add it if it sounds better.